crates/sui-framework/packages/dwallet-system/sources/ethereum_state.move

```diff
@@ -99,21 +103,30 @@ module dwallet_system::ethereum_state {
         assert!(latest_ethereum_state_id == object::id(latest_ethereum_state), EStateObjectMismatch);
 
         let eth_state_bytes = get_ethereum_state_data(eth_state);
-        let (data, time_slot, network) = verify_eth_state(
+        let (data, time_slot, network, state_root) = verify_eth_state(
             updates_bytes,
             finality_update_bytes,
             optimistic_update_bytes,
             eth_state_bytes,
+            beacon_block,
+            beacon_block_body,
+            beacon_block_execution_payload,
+            beacon_block_type,
         );
 
         assert!(network == latest_ethereum_state.network, ENetworkMismatch);
-        assert!(time_slot > latest_ethereum_state.last_slot, EUpdateIrrelevant);
+        assert!(time_slot >= latest_ethereum_state.last_slot, EUpdateIrrelevant);
+
+        if(time_slot == latest_ethereum_state.last_slot) {
+            return
+        };
```

Comment:
```suggestion
        assert!(time_slot >= latest_ethereum_state.last_slot, EUpdateIrrelevant);
```

Reply:
This is incorrect.
If the time slot is the same then I want the transaction to return without an error.

Reply:
The >= will catch also the case it is ==. The if block will never get executed.

Reply:
this is an assert statement, means if it is NOT >=, then it would fail and abort.
if it is equal or bigger than, it would continue

Reply:
gotcha my bad